### PR TITLE
Always check rank in View construction 

### PR DESCRIPTION
**Pull request by masterleinad**
https://github.com/kokkos/kokkos/pull/4618 introduced checks for the rank in View constructors except that it disabled them for the constructor that takes a `view_alloc`  argument. After #5188 and a minor fix, this should not be necessary anymore and we can just get rid of the option.
Related to https://github.com/arborx/ArborX/pull/758.

### CODE REVIEW

File: core/src/Kokkos_CopyViews.hpp [hunk 1]
```diff
@@ -3533,24 +3533,10 @@ create_mirror(const Kokkos::View<T, P...>& src,
...
-  layout.stride[2] = src.stride_2();
-  layout.stride[3] = src.stride_3();
-  layout.stride[4] = src.stride_4();
-  layout.stride[5] = src.stride_5();
-  layout.stride[6] = src.stride_6();
-  layout.stride[7] = src.stride_7();
+  for (unsigned int i = 0; i < src_type::rank; ++i) {
+    layout.dimension[i] = src.extent(i);
+    layout.stride[i]    = src.stride(i);
+  }
```
Comment (PhilMiller) [inline core/src/Kokkos_CopyViews.hpp:3539]: Would there be something wrong with just `Kokkos::LayoutStride layout = src.layout();`, or just passing `src.layout()` as an argument and forgoing the copy? We know from the `enable_if` that the source layout is `LayoutStride`

Comment (masterleinad) [reply core/src/Kokkos_CopyViews.hpp:3539]: No, I think we don't even need to specialize for `LayoutStride`.

**Review by dalg24 — APPROVED** [review]

**Comment by janciesko** [comment]
This looks good to me. I believe rebasing this PR vs this #4721 is slightly less work, so I'd prefer to merge #4721 first in case both PRs start to pass CI.

**Review by janciesko — APPROVED** [review]

**Comment by dalg24** [comment]
> This looks good to me. I believe rebasing this PR vs this #4721 is slightly less work, so I'd prefer to merge #4721 first in case both PRs start to pass CI.

I thought about it and
* This PR is ready and #4722 is not
* Merging this one first will reduce the diff in #4722 
* The conflicts in `Kokkos_CopyViews.hpp` are manageable

Therefore I am merging this one first

**Comment by dalg24** [comment]
I took resolved the conflicts in the other PR

**Pull request merged**